agent/mibgroup/hardware/fsys/fsys_mntent.c

```diff
@@ -176,7 +179,240 @@ _fsys_type( char *typename )
 void
 netsnmp_fsys_arch_init( void )
 {
-    return;
+    snmpd_register_config_handler("ignoremount", parse_mount_config,
+                                  free_mount_config, "name");
+}
+
+#define ITEM_STRING	1
+#define ITEM_SET	2
+#define ITEM_STAR	3
+#define ITEM_ANY	4
+
+typedef unsigned char details_set[32];
+
+typedef struct _conf_mount_item {
+    int             item_type;  /* ITEM_STRING, ITEM_SET, ITEM_STAR, ITEM_ANY */
+    void           *item_details;       /* content depends upon item_type */
+    struct _conf_mount_item *item_next;
+} conf_mount_item;
+
+typedef struct _conf_mount_list {
+    conf_mount_item *list_item;
+    struct _conf_mount_list *list_next;
+} conf_mount_list;
+static conf_mount_list *conf_list = NULL;
+
+static int      match_mount_config(const char *);
+static int      match_mount_config_item(const char *, conf_mount_item *);
+
+static void
+parse_mount_config(const char *token, char *cptr)
+{
+    conf_mount_list *d_new = NULL;
+    conf_mount_item *di_curr = NULL;
+    details_set    *d_set = NULL;
+    char           *name = NULL, *p = NULL, *d_str = NULL, c;
+    unsigned int    i, neg, c1, c2;
+    char           *st = NULL;
+
+    name = strtok_r(cptr, " \t", &st);
+    if (!name) {
+        config_perror("Missing NAME parameter");
+        return;
+    }
+    d_new = (conf_mount_list *) malloc(sizeof(conf_mount_list));
```

Comment:
Please do not cast the malloc() return value. Such casts are not necessary since the return type of malloc() is `void *`.